libraries/mbed/common/rtc_time.c

```diff
@@ -58,6 +75,15 @@ clock_t clock() {
     return t;
 }
 
+void attach_rtc(time_t (*read_rtc)(void), void (*write_rtc)(time_t), void (*init_rtc)(void), int (*isenabled_rtc)(void)) {
+    _rtc_read = read_rtc;
+    _rtc_write = write_rtc;
+    _rtc_init = init_rtc;
+    _rtc_isenabled = isenabled_rtc;
+}
```

Comment:
This should not be an atomic operation?
So if I call this function, yes, I "switch RTC callbacks" to new ones but where is the code to for example:
* force new RTC re-initialization. 
My point is I only call new ```rtc_init()``` when I set new time. Maybe I should have some flag saying I changed RTC API callbacks, just in case if someone calls rtc read/write I will reinit for him.

I probably missed something but I think some piece of comment  / documentation about (example usage of: ```attach_rtc()```) how it should work in source code would be great.